core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -239,20 +248,26 @@ public Object getFieldValue(MasterInfo o) {
       MasterUtils.createMasters(mJournalSystem, mRegistry, mSafeModeManager);
 
       // Create config checker
-      mMasterConfigChecker = new ServerConfigurationChecker();
-      mWorkerConfigChecker = new ServerConfigurationChecker();
+      mMasterConfigRecord = new ServerConfigurationRecord();
+      mWorkerConfigRecord = new ServerConfigurationRecord();
+      mConfigChecker = new ServerConfigurationChecker(mMasterConfigRecord, mWorkerConfigRecord);
+      mConfigChecker.registerGetMasterHostnameListener(this::getMasterHostnameHandler);
+      mConfigChecker.registerGetWorkerHostnameListener(this::getWorkerHostnameHandler);
```

Comment:
when we register masters and workers with the config checker, can we include the hostname at that time? Then we don't need to pass a method here

Reply:
In ServerConfigurationRecord, instead of recording the node id, we change to record the node hostname.

```
private final Map<String, List<ConfigRecord>> mConfMap;
```